proto/kv/v1/kv.proto

```diff
@@ -49,8 +65,42 @@ message KeyValuePutResponse {}
 message KeyValueDeleteRequest {
   // The collection containing the existing keyValue to be deleted
   string collection = 1;
+
   // The unique key of the keyValue to delete
-  string key = 2;
+  map<string, Key> key = 2;
 }
 
-message KeyValueDeleteResponse {}
\ No newline at end of file
+message KeyValueDeleteResponse {}
+
+message KeyValueQueryRequest {
+  message Expression {
+    // The query operand or attribute
+    string operand = 1;
+
+    // The query operator [ = | < | <= | > | >= | startsWith ]
+    string operator = 2;
+
+    // The query expression value
+    string value = 3;
+  }
+
+  // The query from collection
+  string collection = 1;
+
+  // The query expressions
+  repeated Expression expressions = 2;
+
+  // The query fetch limit
+  int32 limit = 3;
+
+  // The query paging continuation token
+  map<string, Key> pagingToken = 4;
+}
+
+message KeyValueQueryResponse {
+  // The retrieved values
+  repeated google.protobuf.Struct values = 1;
+
+  // The query paging continuation token, when not null further results are available
+  map<string, Key> pagingToken = 2;
```

Comment:
Have been thinking about pagination and streaming, we can still support unary request style queries, but to save some burden in the SDKs we might actually be able to ask the membrane to take care of the paging for us using something like:
```proto
  // In this case we would no longer require the return of a paging token 
  rpc Query (KeyValueQueryRequest) returns (stream KeyValueQueryResponse);
```
In end user runtime we would be able to handle a stream without having to deal with paging at all, and the membrane just moves onto the next page as the use continues to read from the stream, eventually sending an EOF once its done writing.

Reply:
Agree it would make the SDK simpler. However there are UX where you want pagination, as you step through a step through pages of results.